File: app/src/main/java/com/alexkn/syntact/presentation/hangman/ListItemViewHolder.java
package com.alexkn.syntact.presentation.hangman;

import android.view.View;

import com.alexkn.syntact.domain.common.Identifiable;

import androidx.recyclerview.widget.RecyclerView;

public abstract class ListItemViewHolder<T extends Identifiable> extends RecyclerView.ViewHolder {
    public ListItemViewHolder(View v) {
        super(v);
    }

    public abstract void bindTo(T entity);
}

File: app/src/main/java/dev/alexknittel/syntact/presentation/common/ListItemViewHolder.kt
package dev.alexknittel.syntact.presentation.common

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import dev.alexknittel.syntact.core.model.Identifiable

abstract class ListItemViewHolder<T : Identifiable<*>>(v: View) : RecyclerView.ViewHolder(v) {

    abstract fun bindTo(entity: T)
}
